### PR TITLE
longest_common_prefix_java code

### LINKED ISSUE #2042: longest common prefix
**Is your feature request related to a problem? Please describe.**
The Algorithm for longest common prefix in java is not there.

**Describe the solution you'd like**
Like to add the code in java for the same.

**Do you want to work on it**
I want to work in java language.

Please assign me this issue.
@rish-singhal

### PR DESCRIPTION
**Pull request by ssaumyaa7**
This is the java code for longest common prefix @plazzy99 @rish-singhal 

Closes #2042

### CODE REVIEW

**Comment by rish-singhal** [comment]
Resolve conflicts. Sort the readme list & give spaces around operators =,>,<,... if they are not there.

**Comment by ssaumyaa7** [comment]
@rish-singhal

File: Code/Java/Longest_Common_Prefix.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        String[] words = str.split("\\s+");
+		System.out.println(longestCommonPrefix(words));
+	}
+}
+
+
+/*
+
+    Test case:
+        Input:  small smile smell
```
Comment (rish-singhal) [inline Code/Java/Longest_Common_Prefix.java:51]: Give atleast 2 test cases. Shift time complexity down

File: Code/Java/Longest_Common_Prefix.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                }
+            }
+            lcp+=c;
+            index++;
+        }
+        return lcp;
+        
+    }
+    public static void main(String[] args) {
+        Scanner sc= new Scanner(System.in);
```
Comment (rish-singhal) [inline Code/Java/Longest_Common_Prefix.java:40]: Give spaces around =, and at other places too

File: string/readme.md [hunk 3]
```diff
@@ -29,5 +29,7 @@ A string is a class that contains a char array, but automatically manages it for
 * Revere a string using Stack ---->[C++](/Code/C++/reverse_a_string_using_stack.cpp)
 * Reverse Individual Words in String ----> [Java](/Code/Java/revindivstring.java)
 * Maximum occuring character in a string ---->[C++](/Code/C++/max_count.cpp)
```
Comment (rish-singhal) [inline string/readme.md:31]: Sort this list

File: Code/Java/Longest_Common_Prefix.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        
+        for(char c:strs[0].toCharArray())
+        {
+            for(int i=1;i<strs.length;i++)
+            {
+                if(index>= strs[i].length() || c!=strs[i].charAt(index)){
+                    return lcp;
+                }
+            }
+            lcp+=c;
```
Comment (rish-singhal) [inline Code/Java/Longest_Common_Prefix.java:33]: Give space here

File: Code/Java/Longest_Common_Prefix.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+Takes constant space.
+
+_______________________________________________________________
+*/ 
+
+import java.util.Scanner; 
+
+public class Main {
+    private static String longestCommonPrefix(String[] strs) {
+        String lcp="";
```
Comment (rish-singhal) [inline Code/Java/Longest_Common_Prefix.java:17]: Here too

File: Code/Java/Longest_Common_Prefix.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        
+        if(strs.length==0 || strs==null){
```
Comment (rish-singhal) [inline Code/Java/Longest_Common_Prefix.java:19]: Here too around ==

**Review by rish-singhal — CHANGES_REQUESTED** [review]

File: string/readme.md (hunk 3 above)
Comment (ssaumyaa7) [reply string/readme.md:31]: sort list as in how?

Comment (rish-singhal) [reply string/readme.md:31]: The list should be in alphabetical order, A, b , c,...

**Comment by rish-singhal** [comment]
Resolve conflicts

**Comment by ssaumyaa7** [comment]
Resolved

File: Code/Java/Longest_Common_Prefix.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        if(strs.length == 0 || strs == null){
+            return lcp;
+        }
+        
+        int index = 0;
+        
+        for(char c:strs[0].toCharArray())
+        {
+            for(int i=1;i<strs.length;i++)
+            {
```
[the remark this answers is not in the document]
Comment (rish-singhal) [reply Code/Java/Longest_Common_Prefix.java:22]: still not updated

File: Code/Java/Longest_Common_Prefix.java [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    private static String longestCommonPrefix(String[] strs) {
+        String lcp = "";
+        
+        if(strs.length == 0 || strs == null){
+            return lcp;
+        }
+        
+        int index = 0;
+        
+        for(char c:strs[0].toCharArray())
```
[the remark this answers is not in the document]
Comment (rish-singhal) [reply Code/Java/Longest_Common_Prefix.java:19]: still not updated

File: string/readme.md [hunk 9]
```diff
@@ -23,11 +23,12 @@ A string is a class that contains a char array, but automatically manages it for
...
-* Longest Palindromic Substring ----> [Python](/Code/Python/longest_palindromic_substring.py)
 * Longest Prefix Suffix Problem ---->[C++](/Code/C++/longest_prefix_suffix.cpp)
 * Manacher's algorithm (finding the longest palindrome in a string) ----> [Java](/Code/Java/longPalindrome.java)
 * Maximum occuring character in a string ---->[C++](/Code/C++/max_count.cpp)
 * Number of deletions to make a string pallindrome ---->[Python](/string/Number_of_deletions_to_make_pallindrome.py)
 * Reverse a string ---->[C++](/Code/C++/reverse_string.cpp) | [Java](/Code/Java/reverse_string.java)
-* Revere a string using Stack ---->[C++](/Code/C++/reverse_a_string_using_stack.cpp)
+* Reverse a string using Stack ---->[C++](/Code/C++/reverse_a_string_using_stack.cpp)
 * Reverse Individual Words in String ----> [Java](/Code/Java/revindivstring.java)
+* Longest Common Prefix ----> [Java](/Code/Java/Longest_Conmmon_Prefix.java)
```
Comment (rish-singhal) [inline string/readme.md:33]: this list is not sorted now.

**Review by rish-singhal — CHANGES_REQUESTED** [review]

**Comment by Rohan27s** [comment]
@rish-singhal check the changes requested by you

File: Code/Java/Longest_Common_Prefix.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (rish-singhal) [reply Code/Java/Longest_Common_Prefix.java:22]: Space should be after ; not before... so it should be int i = 1;

File: Code/Java/Longest_Common_Prefix.java [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+Space complexity:
+Takes constant space.
+*/
+
+/*
+
+    Test case:
+    
+        Input:  small smile smell
+        Output: sm
```
Comment (rish-singhal) [inline Code/Java/Longest_Common_Prefix.java:55]: Align this correctly as below test case. (Improve indentation)

**Comment by Rohan27s** [comment]
@ssaumyaa7 make these changes before 31/05 to get this PR marked under GSSOC

**Comment by ssaumyaa7** [comment]
@Rohan27s @rish-singhal

**Comment by rish-singhal** [comment]
Which issue is this PR for?

File: Code/Java/Longest_Common_Prefix.java [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+*/ 
+
+import java.util.Scanner; 
+
+public class Main {
+    private static String longestCommonPrefix(String[] strs) {
+        String lcp = "";
+        
+        if(strs.length == 0 || strs == null)
+	{
```
Comment (rish-singhal) [inline Code/Java/Longest_Common_Prefix.java:14]: Align this { properly!

File: Code/Java/Longest_Common_Prefix.java [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+        }
+        
+        int index = 0;
+        
+        for(char c : strs[0].toCharArray())
+        {
+            for(int i=1 ; i<strs.length ; i++)
+            {
+                if(index >= strs[i].length() || c != strs[i].charAt(index))
+		{
```
[the remark this answers is not in the document]
Comment (rish-singhal) [reply Code/Java/Longest_Common_Prefix.java:25]: This is not aligned still, please align { properly

File: Code/Java/Longest_Common_Prefix.java [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+                    return lcp;
+                }
+            }
+            lcp += c;
+            index++;
+        }
+        return lcp ;
+        
+    }
```
Comment (rish-singhal) [inline Code/Java/Longest_Common_Prefix.java:34]: Give empty line here before the declaration of function

File: Code/Java/Longest_Common_Prefix.java [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+    public static void main(String[] args) 
+    {
+        Scanner sc = new Scanner(System.in);
+        String str = sc.nextLine();
+        String[] words = str.split("\\s+");
+	System.out.println(longestCommonPrefix(words));
```
Comment (rish-singhal) [inline Code/Java/Longest_Common_Prefix.java:40]: Indent this line well

**Review by rish-singhal — CHANGES_REQUESTED** [review]

**Comment by ssaumyaa7** [comment]
gitcode
I don't know what issue is there, from my end it is indented well. Please check.

**Comment by rish-singhal** [comment]
Hey also give extra line before function declaration.

**Comment by rish-singhal** [comment]
That's weird let me see again on GitHub.

**Comment by rish-singhal** [comment]
Screenshot 2021-05-29 at 4 31 39 PM
the two brackets are not properly aligned.

**Pull request merged**